dev/NavigationView/readme.md

```diff
@@ -0,0 +1,51 @@
+# NavigationView Control
+
+NavigationView is a powerful control that makes it easy to integrate a rich navigation experience into apps.
+This document outlines overall structure and behaviors.
+Information related to the NavigationViewItem, or the rendering of the control see the documents in the /NavigationView/docs folder.
+
+## Architecture
+
+NavigationView is a complex control, which has a lot of internal states to store.
+
+Rendering of the menu items is being done through three ItemsRepeaters, one for the NavigationView in left mode, one for the NavigationView in top mode and one for the overflow menu.
+For more on this, see [the rendering document](./docs/rendering.md).
+
+The selection is being tracked through a SelectionModel, which uses IndexPath tracking for hierarchical NavigationViewItems.
+Every NavigationViewItem can be selected by either setting the `IsSelected` property of a NavigationViewItem, or setting the selected item.
```

Comment:
Might be helpful to explicitly define that "setting the selected item" is a property on NavigationView (and also add the item invoke selection path for the sake of being thorough)

Selection can be achieved via 3 paths:
- `NavigationViewItem` invoke (user interaction)
- `NavigationViewItem` IsSelected property (API) 
- `NavigationView` SelectedItem property (API)

Reply:
Good point, switched to your list.